### DISCUSSION TITLE
Extracting reponse content for subsequent requests is very verbose, am I missing a convenience syntax?

### DISCUSSION THREAD

**Discussion by ppenguin** [rest-nvim/rest.nvim#489] [Q&A]
Just wondering if there's no (much) more compact way to express the following:

```
# Get free
GET {{base}}/getmytoken

# @lang=lua
> {%
local body = vim.json.decode(response.body)
client.global.set("regtoken", body.regtoken)
%}

### test contents of response
POST https://httpbin.org/post HTTP/1
Content-Type: application/json

{
  "regtoken": "{{regtoken}}"
}
```

For that matter, I also tried without `@lang=lua` which, according to the spec, would default to javascript, but this didn't work, presumably because the lua env has no access to a js interpreter?

But anyway, I would expect there to be an automatic variable containing the last response which could be used directly?

**Comment by boltlessengineer** [comment]
Sorry for inconvenience but that's the most compact way to extracting response content now.

`rest.nvim` aims to be compatible with [jetbrain's http syntax](https://www.jetbrains.com/help/idea/exploring-http-syntax.html) but nothing else.
And this syntax doesn't support automatic response decoding afaik.

By spec, scripts will default to javascript but `rest.nvim` currently don't have ability to process the JS code. It is definitely a future goal though!

**Reply by ppenguin** [reply]
Thanks for the info. Personally I wouldn't care much for JS when we already have lua (I was just trying to understand which part of the spec applies to the plugin), but more "automatic" response handling at some point would be really cool.
I'm not sure whether one could consider sort of "hook functions" that could be called to keep backward compatibility with jetbrain and have other things just as an add on that the nvim plugin adds.

**Comment by chardskarth** [comment]
A PR by some awesome dude is on the way: https://github.com/rest-nvim/rest.nvim/pull/516
